src/components/NavBar/NavBar.tsx

```diff
@@ -15,7 +20,23 @@ interface NavBarProps {
 }
 
 const NavBar = ({ pathOptions, endButton, headerImage }: NavBarProps) => {
+  const [userName, setUserName] = useState<string>("");
+
   const location = useLocation();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const token = getToken();
+    if (token) {
+      const decodedToken = JSON.parse(atob(token.split(".")[1]));
+      setUserName(decodedToken.username);
+    }
+  }, []);
```

Comment:
Decoding and storing a token might be better suited to be processed in a React Context - maybe an `AuthContext`, which will get the user info and from the token and make it accessible for any component that needs it in the future.
At the moment you are only setting the username, but in the future you might have roles and permissions encoded into the token and will possibly display/hide certain features based on that.

Reply:
A good example can be
https://dev.to/dayvster/use-react-context-for-auth-288g